modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/ContentCompressionDecompressionHandler.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.wso2.andes.kernel.disruptor;
+
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Factory;
+import net.jpountz.lz4.LZ4FastDecompressor;
+import org.wso2.andes.configuration.AndesConfigurationManager;
+import org.wso2.andes.configuration.enums.AndesConfiguration;
+import org.wso2.andes.kernel.AndesMessagePart;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This singleton class handles compression and decompression of AMQP message contents
+ */
+public class ContentCompressionDecompressionHandler {
+
+    private static ContentCompressionDecompressionHandler instance;
+
+    /**
+     * Whether message content compression from the server side is enabled
+     */
+    public static Boolean isCompressionEnabled = AndesConfigurationManager.readValue
+            (AndesConfiguration.PERFORMANCE_TUNING_ALLOW_COMPRESSION);
+
+    /**
+     * Maximum content size to check before compress message content, in bytes. Messages with
+     * content less than this size will not compress.
+     */
+    public static int contentCompressionThreshold = AndesConfigurationManager.readValue
+            (AndesConfiguration.PERFORMANCE_TUNING_CONTENT_COMPRESSION_THRESHOLD);
+
+    /**
+     * Getting lz4 instance
+     */
+    static LZ4Factory factory = LZ4Factory.fastestInstance();
+
+    /**
+     * Getting lz4 fast compressor to compress data
+     */
+    static LZ4Compressor compressor = factory.fastCompressor();
```

Comment:
confirm that using the native (JNI) is platform neutral. otherwise we have to switch to java implementation (from the factory)